sorting/Bogo_Sort.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file
+ * @brief Implementation of [Bogosort algorithm](https://en.wikipedia.org/wiki/Bogosort)
+ *
+ * @details
+ *      In computer science, bogosort (also known as permutation sort, stupid sort, slowsort, 
+ *      shotgun sort, random sort, monkey sort, bobosort or shuffle sort) is a highly inefficient 
+ *      sorting algorithm based on the generate and test paradigm. Two versions of this algorithm 
+ *      exist: a deterministic version that enumerates all permutations until it hits a sorted one,
+ *      and a randomized version that randomly permutes its input.Randomized version is implemented here. 
+ *
+ * Algorithm -      
+ *      
+ *      Shuffle the array untill array is sorted.
+ *
+ */
+#include <iostream>
+#include <algorithm>
+#include <array>
+#include <cassert>
+
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * Implement randomized Bogosort algorithm and sort the elements of a given array.
+ * @tparam T typename of the array
+ * @tparam N length of array
+ * @param arr array to sort
+ * @returns new array with elements sorted from a given array
+ */
+template<typename T, size_t N>
+std::array <T, N> randomized_bogosort(std::array <T, N> arr)
+{
+    //Untill array is not sorted
+    while(!std::is_sorted(arr.begin(), arr.end()))
```

Comment:
Shouldn't this be like:

```c++
while (!std::is_sorted(arr.begin(), arr.end())) {

...
```
Same in other parts of the code.

Reply:
The clang-tidy fixes the code. 
@imdeep2905 please enable github actions on your repo and ensure that the branch and repo are up-to-date